### DISCUSSION TITLE
Is there a way to configure fzf's "fuzziness"

### DISCUSSION THREAD

**Discussion by ivan94fi** [junegunn/fzf#3210] [Q&A]
I like fzf for what it is, a _fuzzy_ finder. However, from times to times I find that fzf returns too many matches, with some totally unrelated strings. For example while searching for `sudo` in bash history, one of the matches is:
```
diff 10-quirks.conf /usr/share/X11/xorg.conf.d/10-quirks.conf
             ^       ^                       ^            ^
```
where the `^` symbols represent the highlighted matched characters.

I found the `--exact` configuration option, but this is too strict: I would like to tell fzf to ignore matches that are too different from the query, but keep the ones not too different.

I am thinking of some parameter corresponding to a threshold value on the distance of the returned string from the original query string, see for example [the threshold option](https://fusejs.io/api/options.html#threshold) in Fuse.js, a Javascript fuzzy search library.

A threshold of 0 would mean to return an exact match, a threshold of 1 (or 100) would mean be as fuzzy as possible, show every single matching result, no matter how far from the query string.

Is this possible in fzf?

Thank You,
Ivan

**Comment by iolpltuisciaoln** [comment]
Same thoughts.
I think it would be better experience if fzf could first return exact matches, then fuzzy.

**Reply by iolpltuisciaoln** [reply]
i'm encountering this problem using fzf with man
![image](https://github.com/junegunn/fzf/assets/63096974/65fc13f5-6884-4904-937d-9d2fecd013eb)
![image](https://github.com/junegunn/fzf/assets/63096974/4ce9b026-b5cf-4040-86c5-119125b4718c)
![image](https://github.com/junegunn/fzf/assets/63096974/ced6f586-e12d-4388-af54-e48a0bca1d0c)
